dev/dev_setup.sh

```diff
@@ -0,0 +1,306 @@
+                $MACOS12 | $MACOS14)
+                    notice "Installing unzip via Homebrew"
+                    logged brew install unzip && USED_BREW=true;;
+                $UBUNTU20 | $UBUNTU22)
+                    notice "Installing unzip via apt-get"
+                    logged apt-get install -y unzip;;
+                *) unreachable "Unsupported platform; did not install unzip"; return;;
+            esac
+        fi
+        logged unzip solvers.bin.zip
+
+        # If we want to install more solvers by default, we can do so here
+        for solver in cvc4 cvc5 z3; do
+            if ! is_installed $solver ; then
+                notice "Installing $solver"
+                chmod u+x $solver
+                # Try installing without sudo first, because some environments
+                # don't have it installed by default
+                mv $solver /usr/local/bin || sudo mv $solver /usr/local/bin
+            fi
+        done
+
+        cd $HERE
+        rm -r $solvers_dir
+    else
+        notice "Not installing cvc4, cvc5, or z3 solvers because they already exist"
+    fi
+
+    # Make sure the installed versions are the versions that have been tested
+    check_version cvc4 $WHAT4_CVC4_VERSION
+    check_version cvc5 $WHAT4_CVC5_VERSION
+    check_version z3 $WHAT4_Z3_VERSION
+}
+
+# Checks the version of the given command
+# Usage: check_version <command> <expected version>
+#
+# Assumes that the command can be called with `--version`
+check_version() {
+    cmd="$1"
+    expected="$@"
+    version_file=$(mktemp)
+    $cmd --version > $version_file
+    if ! (grep -iq "$expected" $version_file); then
+        actual_version=$(grep -i -m 1 "version" $version_file)
+        notice "Your version of $cmd is unexpected; expected $expected"
+        notice "    Got: $actual_version"
+        notice "    To ensure compatibility, you might want to uninstall the "\
+            "existing version and re-run this script."
+    fi
+    rm $version_file
+}
+
+
+put_brew_in_path() {
+    if $USED_BREW; then
+        # `brew --prefix` is different on macOS 12 and macOS 14
+        echo "export CPATH=$(brew --prefix)/include" >> $VAR_FILE
+        echo "export LIBRARY_PATH=$(brew --prefix)/lib" >> $VAR_FILE
+    fi
+}
+
+set_ubuntu_language_encoding() {
+    case $CRYPTOL_PLATFORM in
+        $UBUNTU20 | $UBUNTU22)
+            notice "Adding language environment variables to $VAR_FILE"
+            echo "export LANG=C.UTF-8" >> $VAR_FILE
+            echo "export LC_ALL=C.UTF-8" >> $VAR_FILE;;
+        *) ;;
+    esac
+}
+
+# Make sure script is running on a supported platform
+notice "Checking whether platform is supported"
+CRYPTOL_PLATFORM=$(supported_platform)
+if [ -z "$CRYPTOL_PLATFORM" ]; then
+    echo "Unsupported platform; this script supports Ubuntu 20.04, Ubuntu 22.04, and macOS 14"
+    exit 1
+fi
+
+update_submodules
+install_ghcup
+install_gmp
+install_zlib
+install_solvers
+put_brew_in_path
+set_ubuntu_language_encoding
```

Comment:
This should probably be moved to the Dockerfile and we shouldn't be overriding the local user's locale settings.  

Maybe it's worth warning the user if `LANG` or `LC_x` doesn't (case-insensitively?) contain "UTF-8" for their local environment, but other than that I think we shouldn't mess with the user's environment like this and just move it to the Dockerfile where we know it needs to be set to support UTF-8.

Also, the conventional wisdom is that you shouldn't set `LC_ALL`: it's intended as something the user can manually set as an override for all of the other `LC_x` settings.  Maybe we decide we're OK with that for the constrained Docker environment, but there's probably a more direct `LC_x` setting (or just set all except `LC_ALL`) that is still effective.

Reply:
I agree we should be careful with this, however another use case for this is to quickly set up VMs. Perhaps a warning + option to set language encoding? We can start handling optional flags for some things like this as well, but ideally we would have a 1 line of code (with possible flags) for setting up a VM as well.

Reply:
Thanks, I did not know this wisdom! I played around a bit and it seems like just setting `LANG` is enough to get the tests to pass. I'll modify this section to check and print a warning, and to add the `export LANG...` line as a comment in the generated var file. That way someone knows there's a problem, can easily uncomment the line if they don't care about changing their settings, but the default behavior lets them decide how to solve the problem.

I'll also update the dockerfile to set these.